### PR TITLE
First Commit

### PR DESCRIPTION
**Pull request by shubhamk840**

### CODE REVIEW

File: Assignment-1/main.swift [hunk 1]
```diff
@@ -0,0 +1,179 @@
+
+import Foundation
+
+class ItemDetails{
+     var name: String
+     var price: Double
+     var quantity: Int
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: The data type for quantity can be optional Int

File: Assignment-1/main.swift [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                    quantity = isItCommand(inputString:inputParts[index+1]) ? 1 : Int(inputParts[index+1])
+                case "-type":
+                    type = isItCommand(inputString:inputParts[index+1]) ? "" : inputParts[index+1]
+                default:
+                    continue
+                }
+            
+        }
+        // checking if all the inputs are valid.
+        if(name==""){
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift:108]: use name as optional , instead of initialising it to empty string by default

File: Assignment-1/main.swift [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            print("Name not Entered")
+        }
+        else if(type==""){
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: same here, as above

File: Assignment-1/main.swift [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            print("Type not Entered")
+        }
+        else if(price == -1){
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: same here , as above

File: Assignment-1/main.swift [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+func inputItems(){
+    
+    var price:Double?
+    var name:String?
+    var type:String?
+    var quantity:Int? = 1
+    
+    func isItCommand(inputString:String)-> (Bool){  // return true if current string is a command else returns false
+        if(inputString=="-name"){//
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Keep Strings in a Seperate String constant file

Comment (shubhamk840) [reply Assignment-1/main.swift]: okay

File: Assignment-1/main.swift [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        else if(type == nil){
+            print("Type not Entered")
+        }
+        else if(price == nil){
+            print("Price Not Entered")
+        }
+        else{
+           // classifying objects on the basis of their type and appending into an array of their type (list)
+            if(type == "imported"){
+                let importedItem = ImportedItem(name!, price! , quantity ?? 1, type!)
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Don't use force unwrapping for optionals,  unwrap it using if-let

File: Assignment-1/main.swift [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+                importedItemList.append(importedItem)
+            }
+            else if(type == "raw"){
+                let rawItem = RawItem(name!, price!, quantity ?? 1, type!)
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Don't use force unwrapping for optionals,  unwrap it using if-let

File: Assignment-1/main.swift [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+                rawItemList.append(rawItem)
+            }
+            else if(type == "manufactured"){
+                let manufacturedItem = ManufacturedItem(name!, price!, quantity ?? 1, type!)
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Don't use force unwrapping for optionals,  unwrap it using if-let

File: Assignment-1/main.swift [hunk 9]
```diff
@@ -0,0 +1,179 @@
+
+import Foundation
+
+class ItemDetails{
+     var name: String
+     var price: Double
+    var quantity: Int? = 1
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift:7]: Ident code,  Select the part of code and press ctrl + I ( Shortcut for identing)

File: Assignment-1/main.swift [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+    var price:Double?
+    var name:String?
+    var type:String?
+    var quantity:Int? = 1
+    
+    func isItCommand(inputString:String)-> (Bool){  // return true if current string is a command else returns false
+        if(inputString=="-name"){//
+            return true
+        }
+        else if(inputString=="-price"){
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Keep Strings in a Seperate String constant file

File: Assignment-1/main.swift [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+            return true
+        }
+        else if(inputString=="-quantity"){
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Keep Strings in a Seperate String constant file

File: Assignment-1/main.swift [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+            return true
+        }
+        else if(inputString=="-type"){
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Keep Strings in a Seperate String constant file

File: Assignment-1/main.swift [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+    }
+    
+    print("Enter the details of the items")
+  
+    if let inputLine = readLine(){
+        let inputParts = inputLine.components(separatedBy: " ") // Breaking the line into string components
+        
+        // considering the input of name differently because it can have more than one word.
+        var indexOfName = 1
+        if(inputParts[0] != "-name"){ // first command must be name
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: before accessing index , put a safety check here. check if inputParts.count > 0 , only then you should access that element

File: Assignment-1/main.swift [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+            print("wrong input format")
+        }
+        else{
+            name=""
+            
+            // using while loop to extract name.
+            while( indexOfName < inputParts.endIndex && !isItCommand(inputString: inputParts[indexOfName])){ // continue looping till we don't hit another command.
+                name! += inputParts[indexOfName] + " ";
```
Comment (AdarshRavikumar) [inline Assignment-1/main.swift]: Don't use force unwrapping, use if - let

**Review by AdarshRavikumar — APPROVED** [review]

File: Assignment-1/main.swift (hunk 6 above)
Comment (AdarshRavikumar) [reply Assignment-1/main.swift]: Still not reolved

File: Assignment-1/main.swift (hunk 7 above)
Comment (AdarshRavikumar) [reply Assignment-1/main.swift]: Still not reolved

File: Assignment-1/main.swift (hunk 8 above)
Comment (AdarshRavikumar) [reply Assignment-1/main.swift]: Still not reolved

**Pull request merged**